File: src/dataset/typescript/protein-translation_exercism-javascript-samajammin/claude/ClaudeSonnet4_protein-translation_v1.ts
class ProteinTranslation {
  static readonly codonProteinMap: { [codon: string]: string } = {
    AUG: 'Methionine',
    UUU: 'Phenylalanine',
    UUC: 'Phenylalanine',
    UUA: 'Leucine',
    UUG: 'Leucine',
    UCU: 'Serine',
    UCC: 'Serine',
    UCA: 'Serine',
    UCG: 'Serine',
    UAU: 'Tyrosine',
    UAC: 'Tyrosine',
    UGC: 'Cysteine',
    UGU: 'Cysteine',
    UGG: 'Tryptophan',
    UAA: 'STOP',
    UAG: 'STOP',
    UGA: 'STOP'
  };

  static proteins(codonString: string): string[] {
    const proteins: string[] = [];
    
    for (let i = 0; i < codonString.length; i += 3) {
      const codon = codonString.slice(i, i + 3);
      if (codon.length < 3) break;
      
      const protein = this.codonProteinMap[codon];
      if (protein === 'STOP') {
        break;
      }
      proteins.push(protein);
    }
    
    return proteins;
  }
}

export default ProteinTranslation;

File: src/dataset/python/protein-translation/claude/ClaudeSonnet4_protein-translation_v1.py
#codons dictionary indicates the matching protein for the codon
codons = {'AUG' : 'Methionine', 'UUU' : 'Phenylalanine',
          'UUC' : 'Phenylalanine', 'UUA' : 'Leucine' ,
          'UUG' : 'Leucine', 'UCU' : 'Serine',
          'UCC' : 'Serine','UCA' : 'Serine', 
          'UCG' : 'Serine', 'UAU' : 'Tyrosine', 
          'UAC' : 'Tyrosine', 'UGU' : 'Cysteine',
          'UGC' : 'Cysteine', 'UGG' : 'Tryptophan'}
#stop set contains the STOP codon segments
stop = frozenset({'UAA', 'UAG', 'UGA'})

def proteins(strand):
    protein = []
    for e in range(0, len(strand), 3):
        codon = strand[e:e+3]
        if codon in stop:
            break
        if codon in codons:
            protein.append(codons[codon])
    return protein